src/Rules/Library/ControlShouldSupportTextPattern.cs

```diff
@@ -28,7 +29,13 @@ public override EvaluationCode Evaluate(IA11yElement e)
 
         protected override Condition CreateCondition()
         {
-            return Document | Edit;
+            var win32Edit = Edit & StringProperties.Framework.Is(Framework.Win32);
+            var nonfocusableDirectUIEdit = Edit & ~IsKeyboardFocusable & StringProperties.Framework.Is(Framework.DirectUI);
+
+            return Document
+                | (Edit
+                & ~win32Edit
```

Comment:
>win32Edit [](start = 19, length = 9)

This executes only once per rule, right? If not, it might be worth making these static to save memory

Reply:
The CreateCondition function is only called once per rule per session.